editor/project_manager.cpp

```diff
@@ -109,245 +110,283 @@ class ProjectDialog : public ConfirmationDialog {
+				while (ret == UNZ_OK) {
+					unz_file_info info;
+					char fname[16384];
+					ret = unzGetCurrentFileInfo(pkg, &info, fname, 16384, nullptr, 0, nullptr, 0);
+					if (ret != UNZ_OK) {
+						break;
 					}
 
-					if (ret == UNZ_END_OF_LIST_OF_FILE) {
-						set_message(TTR("Invalid \".zip\" project file; it doesn't contain a \"project.godot\" file."), MESSAGE_ERROR);
-						get_ok_button()->set_disabled(true);
-						unzClose(pkg);
-						return "";
+					if (String::utf8(fname).ends_with("project.godot")) {
+						break;
 					}
 
+					ret = unzGoToNextFile(pkg);
+				}
+
+				if (ret == UNZ_END_OF_LIST_OF_FILE) {
+					set_message(TTR("Invalid \".zip\" project file; it doesn't contain a \"project.godot\" file."), MESSAGE_ERROR);
+					create_dir->hide();
 					unzClose(pkg);
+					return "";
+				}
 
-					// check if the specified install folder is empty, even though this is not an error, it is good to check here
-					d->list_dir_begin();
-					is_folder_empty = true;
-					String n = d->get_next();
-					while (!n.is_empty()) {
-						if (!n.begins_with(".")) {
-							// Allow `.`, `..` (reserved current/parent folder names)
-							// and hidden files/folders to be present.
-							// For instance, this lets users initialize a Git repository
-							// and still be able to create a project in the directory afterwards.
-							is_folder_empty = false;
-							break;
-						}
-						n = d->get_next();
-					}
-					d->list_dir_end();
+				create_dir->show();
+				unzClose(pkg);
+			} else {
+				set_message(TTR("The path specified is not a valid project."), MESSAGE_ERROR);
+				create_dir->hide();
+				return "";
+			}
+		}
 
-					if (!is_folder_empty) {
-						set_message(TTR("Please choose an empty folder."), MESSAGE_WARNING, INSTALL_PATH);
-						get_ok_button()->set_disabled(true);
-						return "";
-					}
+		if (mode == MODE_NEW || mode == MODE_IMPORT) {
+			if (target_path == working_dir || target_path == OS::get_singleton()->get_executable_path().get_base_dir()) {
+				// Running the engine without arguments will always check the working directory for a project.
+				// If the default working directory of the editor contains a project, users won't be able to open the project manager!
+				set_message(TTR("Creating a project in the engine's working directory or executable directory is not allowed, as it would prevent the project manager from starting."), MESSAGE_ERROR, target_path_type);
+				return "";
+			}
+
+#ifdef WINDOWS_ENABLED
+			const String home_dir = OS::get_singleton()->get_environment("USERPROFILE");
+#else
+			const String home_dir = OS::get_singleton()->get_environment("HOME");
+#endif
+
+			if (target_path == home_dir || target_path == OS::get_singleton()->get_system_dir(OS::SYSTEM_DIR_DOCUMENTS)) {
+				set_message(TTR("You cannot save a project in the selected path. Please make a new folder or choose a new path."), MESSAGE_ERROR, target_path_type);
+				return "";
+			}
```

Comment:
First check: Disallow working directory and executable directory
Second check: Disallow home dir (env USERPROFILE or HOME) and documents dir

Reply:
why are you disallowing the documents folder?!